src/config/settings.rs

```diff
@@ -0,0 +1,140 @@
+use core::fmt::Display;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// A setting can be either a signed integer, unsigned integer, string, map or boolean.
+/// Maps could be created by using comma separated strings maybe
+#[derive(Clone, PartialEq, Debug)]
+pub enum Setting {
+    SInt(isize),
+    UInt(usize),
+    String(String),
+    Bool(bool),
+    Map(Vec<String>),
+}
+
+impl Serialize for Setting {
```

Comment:
serialize / deserialize are needed for using serde_json in the json-adapter. It will convert a Setting directory to a string